src/Sylius/Component/Core/spec/Model/ProductSpec.php

```diff
@@ -67,6 +68,25 @@ function its_taxons_are_mutable(Collection $taxons)
         $this->getTaxons()->shouldReturn($taxons);
     }
 
+    function its_taxons_should_be_returned_by_taxonomy(
+        TaxonInterface $taxon,
+        TaxonInterface $rootTaxon
+    )
+    {
+        $rootTaxon->addChild($taxon);
+        $rootTaxon->setName('Brand');
+
+        $taxon->getRoot()->willReturn($rootTaxon);
+        $taxons = new ArrayCollection([$taxon->getWrappedObject()]);
+
+        $this->setTaxons($taxons);
+        $taxon->getRoot()->shouldBeCalled();
+        $rootTaxon->getName()->shouldBeCalled();
+
+        $this->getTaxons('Brand')
+            ->shouldHaveType('Doctrine\Common\Collections\Collection');
```

Comment:
71  - its taxons should be returned by taxonomy
      expected [obj:Doctrine\Common\Collections\ArrayCollection], but got
      [obj:Doctrine\Common\Collections\ArrayCollection].